sky/utils/control_master_utils.py

```diff
@@ -34,16 +41,53 @@ def is_tmp_9p_filesystem() -> bool:
     return filesystem_types[1].lower() == '9p'
 
 
-@functools.lru_cache
-def should_disable_control_master() -> bool:
+def should_disable_control_master(ip: str) -> bool:
     """Whether disable ssh control master based on file system.
 
+    Args:
+        ip: The ip address of the node.
+
     Returns:
         bool: True if the ssh control master should be disabled,
         False otherwise.
     """
-    if is_tmp_9p_filesystem():
+    if is_unsupported_filesystem():
+        return True
+    if is_high_concurrency(ip):
         return True
     # there may be additional criteria to disable ssh control master
     # in the future. They should be checked here
     return False
+
+
+@functools.lru_cache(maxsize=1)
+def is_unsupported_filesystem():
+    """Determine if the filesystem is unsupported."""
+    return is_tmp_9p_filesystem()
+
+
+def is_high_concurrency(ip: str) -> bool:
+    """Determine if the node has high concurrent ssh connections.
+
+    Args:
+        ip: The IP address to check
+        threshold: Maximum number of allowed concurrent SSH connections
+
+    Returns:
+        bool: True if number of concurrent SSH connections exceeds threshold
+    """
+    try:
+        # Use pgrep to efficiently find ssh processes and pipe to grep for IP
+        cmd = f'pgrep -f ssh | xargs -r ps -p | grep -c {ip}'
```

Comment:
Is that only tracks the SSH command from local machine? How about the connections from other laptops